### PR TITLE
Dont grab twice

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: lib/OpenQA/Scheduler/Scheduler.pm [hunk 1]
```diff
@@ -151,6 +150,11 @@ sub job_grab {
     my $workercaps = $args{workercaps};
 
     my $worker = _validate_workerid($workerid);
+    if ($worker->job) {
+        # TROUBLE ahead - do not call seen, to get into dead worker detection first
```
Comment (aaannz) [inline lib/OpenQA/Scheduler/Scheduler.pm]: I don't understand the comment. Why not to call seen?

Comment (coolo) [inline lib/OpenQA/Scheduler/Scheduler.pm]: because I'm lazy and don't want to duplicate dead worker detection here.

Comment (aaannz) [inline lib/OpenQA/Scheduler/Scheduler.pm]: I'm sorry, I still do not understand. What is harmful on calling `$worker->seen(..)` and doing this check after?

**Comment by okurz** [comment]
LGTM

File: lib/OpenQA/Scheduler/Scheduler.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/Scheduler/Scheduler.pm]: if we update t_updated on every grab, the worker will never get dead - and we won't restart the job

Comment (aaannz) [inline lib/OpenQA/Scheduler/Scheduler.pm]: Now I get it. Thanks.

**Comment by aaannz** [comment]
:+1:

**Pull request merged**